src/EnergyPlus/HVACManager.cc

```diff
@@ -978,145 +978,54 @@ void SimHVAC(EnergyPlusData &state)
-                            }
-                            ShowContinueError(
-                                state, format("Supply-to-demand interface deck 2 temperature check value iteration history trace: {}", HistoryTrace));
-                        }
-                    } // Temps not converged
-
-                    auto &arrayRef4 = state.dataConvergeParams->AirLoopConvergence(AirSysNum).HVACEnergyNotConverged;
-                    if (std::any_of(std::begin(arrayRef4), std::end(arrayRef4), [](bool i) { return i; })) {
-
-                        ShowContinueError(
-                            state,
-                            format("Air System Named = {} did not converge for energy", state.dataAirLoop->AirToZoneNodeInfo(AirSysNum).AirLoopName));
-                        ShowContinueError(state, "Check values should be zero. Most Recent values listed first.");
-                        std::string HistoryTrace = "";
-                        for (int StackDepth = 0; StackDepth < DataConvergParams::ConvergLogStackDepth; ++StackDepth) {
-                            HistoryTrace += format(
-                                "{:.6R},", state.dataConvergeParams->AirLoopConvergence(AirSysNum).HVACEnergyDemandToSupplyTolValue[StackDepth]);
-                        }
-                        ShowContinueError(state, format("Demand-to-Supply interface energy check value iteration history trace: {}", HistoryTrace));
-                        HistoryTrace = "";
-                        for (int StackDepth = 0; StackDepth < DataConvergParams::ConvergLogStackDepth; ++StackDepth) {
-                            HistoryTrace += format(
-                                "{:.6R},", state.dataConvergeParams->AirLoopConvergence(AirSysNum).HVACEnergySupplyDeck1ToDemandTolValue[StackDepth]);
-                        }
-                        ShowContinueError(state,
-                                          format("Supply-to-demand interface deck 1 energy check value iteration history trace: {}", HistoryTrace));
-
-                        if (state.dataAirLoop->AirToZoneNodeInfo(AirSysNum).NumSupplyNodes >= 2) {
-                            HistoryTrace = "";
-                            for (int StackDepth = 0; StackDepth < DataConvergParams::ConvergLogStackDepth; ++StackDepth) {
-                                HistoryTrace +=
-                                    format("{:.6R},",
-                                           state.dataConvergeParams->AirLoopConvergence(AirSysNum).HVACEnergySupplyDeck2ToDemandTolValue[StackDepth]);
-                            }
-                            ShowContinueError(
-                                state, format("Supply-to-demand interface deck 2 energy check value iteration history trace: {}", HistoryTrace));
-                        }
-                    } // energy not converged
-
+                    // mass flow rate
```

Comment:
Future work: create a reference to reuse to avoid these lookups and collapse all these long lines:

```c++
auto & thisSys = state.dataConvergeParams->AirLoopConvergence(AirSysNum);
ConvergenceErrors(state,
                  thisSys.HVACMassFlowNotConverged,
                  thisSys.HVACFlowDemandToSupplyTolValue,
                  thisSys.HVACFlowSupplyDeck1ToDemandTolValue,
                  thisSys.HVACFlowSupplyDeck2ToDemandTolValue,
                  AirSysNum,
                  1);
```

Reply:
@Myoldmopar I tried to use references for all arrays and was unable to pass compile.